dpe/src/context.rs

```diff
@@ -372,32 +374,13 @@ mod tests {
     ///
     /// The context's parent context index is the previous value.
     ///
-    /// So `dpe.contexts[2]` is the parent of `dpe.contexts[4]` which is the parent of
-    /// `dpe.contexts[1]` etc.
-    const CHAIN_INDICES: [usize; MAX_HANDLES] = [
-        2,
-        4,
-        1,
-        13,
-        MAX_HANDLES - 1,
-        3,
-        0,
-        9,
-        5,
-        6,
-        7,
-        8,
-        10,
-        11,
-        12,
-        14,
-        15,
-        16,
-        17,
-        18,
-        19,
-        20,
-        21,
-        22,
-    ];
+    /// So `dpe.contexts[chain_indices[0]]` is the parent of `dpe.contexts[chain_indices[1]]` which is the parent of
+    /// `dpe.contexts[chain_indices[2]]` etc. The chain_indices vector is a random permutation of numbers in [0, MAX_HANDLES).
+    fn get_chain_indices() -> Vec<usize> {
+        let mut chain_indices: Vec<usize> = (0..MAX_HANDLES).collect();
+        let mut rng = rand::thread_rng();
```

Comment:
nit: I think it would be better us use https://docs.rs/rand/0.8.5/rand/trait.SeedableRng.html instead. And then choose a constant as the seed. This will make the tests deterministic. It looks like StdRng in this crate implements `from_seed`: https://docs.rs/rand/0.8.5/rand/rngs/struct.StdRng.html

Reply:
Done